lib/android/app/src/main/java/com/reactnativenavigation/react/ReactView.java

```diff
@@ -12,55 +17,78 @@
-		super(context);
-		this.reactInstanceManager = reactInstanceManager;
-		this.containerId = containerId;
-		this.containerName = containerName;
-		start();
-	}
+    private final ReactInstanceManager reactInstanceManager;
+    private final String containerId;
+    private final String containerName;
+    private boolean isAttachedToReactInstance = false;
 
-	private void start() {
-		setEventListener(new ReactRootView.ReactRootViewEventListener() {
+    private ScrollView scrollView;
+    private ViewTreeObserver.OnScrollChangedListener scrollChangedListener;
 
-			@Override
-			public void onAttachedToReactInstance(final ReactRootView reactRootView) {
-				reactRootView.setEventListener(null);
-				isAttachedToReactInstance = true;
-			}
-		});
-		final Bundle opts = new Bundle();
-		opts.putString("containerId", containerId);
-		startReactApplication(reactInstanceManager, containerName, opts);
-	}
+    public ReactView(final Context context, ReactInstanceManager reactInstanceManager, String containerId, String containerName) {
+        super(context);
+        this.reactInstanceManager = reactInstanceManager;
+        this.containerId = containerId;
+        this.containerName = containerName;
+        start();
+    }
 
-	@Override
-	public boolean isReady() {
-		return isAttachedToReactInstance;
-	}
+    private void start() {
+        setEventListener(reactRootView -> {
+            reactRootView.setEventListener(null);
+            isAttachedToReactInstance = true;
+        });
+        final Bundle opts = new Bundle();
+        opts.putString("containerId", containerId);
+        startReactApplication(reactInstanceManager, containerName, opts);
+    }
 
-	@Override
-	public View asView() {
-		return this;
-	}
+    @Override
+    public boolean isReady() {
+        return isAttachedToReactInstance;
+    }
 
-	@Override
-	public void destroy() {
-		unmountReactApplication();
-	}
+    @Override
+    public View asView() {
+        return this;
+    }
 
-	@Override
-	public void sendContainerStart() {
-		new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidAppear(containerId);
-	}
+    @Override
+    public void destroy() {
+        unmountReactApplication();
+    }
 
-	@Override
-	public void sendContainerStop() {
-		new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidDisappear(containerId);
-	}
+    @Override
+    public void sendContainerStart() {
+        new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidAppear(containerId);
+    }
+
+    @Override
+    public void sendContainerStop() {
+        new NavigationEvent(reactInstanceManager.getCurrentReactContext()).containerDidDisappear(containerId);
+    }
+
+    @Override
+    public void addView(View child, int index, ViewGroup.LayoutParams params) {
+        super.addView(child, index, params);
+        if (child instanceof ScrollView) {
```

Comment:
This means ScrollView must be the root component.

Reply:
We can look for it recursively, but I'm not sure you need to have collapsing Toolbar if ScrollView is not root